### PR TITLE
feat: ✨ verify data path before extracting ID

### PR DESCRIPTION
**Pull request by martonvago**
## Description

This PR makes sure that every time we extract the ID from the data path of a resource, we verify that it is well formed ([discussion](https://github.com/seedcase-project/seedcase-sprout/pull/784#discussion_r1801271054)). 

Related to #784

This PR needs an in-depth review.

## Checklist

- [x] Added or updated tests
- [x] Tests passed locally
- [x] Linted and formatted code
- [x] Build passed locally
- [x] Updated documentation

### CODE REVIEW

File: sprout/core/verify_data_path.py
Comment (martonvago) [inline sprout/core/verify_data_path.py:1]: I moved this to its own file

File: sprout/core/verify_data_path.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    Args:
+        properties: The resource properties to check.
+
+    Returns:
+        The properties, if the data path is well formed.
+
+    Raises:
+        NotPropertiesError: If the data path is not well formed.
+    """
+    data_path = Path(properties.get("path", ""))
```
Comment (martonvago) [inline sprout/core/verify_data_path.py:20]: Changed this to handle the case where `path` is missing

File: tests/core/test_write_resource_properties.py [hunk 2]
```diff
@@ -136,17 +136,42 @@ def test_throws_error_if_resource_properties_are_incorrect(package_properties_pa
...
+    resource_properties_1.path = str(Path("no", "id"))
+    package_properties = PackageProperties(
+        name="my-package",
+        id="123-abc-123",
+        title="My Package",
+        description="This is my package.",
+        version="2.0.0",
+        created="2024-05-14T05:00:01+00:00",
+        resources=[resource_properties_1],
+    )
```
Comment (signekb) [inline tests/core/test_write_resource_properties.py:165]:
```suggestion
    package_properties = PackageProperties(
        name="my-package",
        id="123-abc-123",
        title="My Package",
        description="This is my package.",
        version="2.0.0",
        created="2024-05-14T05:00:01+00:00",
        resources=[str(Path("no", "id"))],
    )
```
Why not just add the malformed path directly?

**Review by signekb — CHANGES_REQUESTED** [review]
Seems fine to me 👍 I have added some minor suggestions

File: sprout/core/verify_data_path.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    if len(data_path.parts) != 3 or not data_path.parts[1].isdigit():
+        error = ResourceError(
```
[the remark this answers is not in the document]
Comment (martonvago) [reply sprout/core/verify_data_path.py:22]: This is a class from `frictionless`, so it comes as is :)

**Review by lwjohnst86 — APPROVED** [review]
:grin: :tada: :clinking_glasses:

**Pull request merged**